bs.py

```diff
@@ -207,14 +204,14 @@ def store_hit(board, ships, guess):    ##TEST THIS BEFORE USING - TEST HITS INCR
 def guess(board, ships):
     while True:
         guess = get_guess(board)
-        os.system('clear')
-        if check_guess_legal(board, guess) == True:
-            if check_cell_guessed(board, guess) == False:
-                if check_guess_hit(board, guess) == True:
+        os.system('clear')  # clever
+        if check_guess_legal(board, guess):  # no need to compare against explicit True/False
+            if not check_cell_guessed(board, guess):
+                if check_guess_hit(board, guess):
                     print "Hit!"
                     board, ships = store_hit(board, ships, guess)
                     break
-                elif check_guess_hit(board, guess) == False:
```

Comment:
Which errors could there be though, and how would calling that function a second time expose them?

You've already called the function once (in the "if" part), you don't need to do it again for the else. There are 2 states that can be returned, "True" or "False". The function can't return anything else, right? Sometimes a function might also return "None" so it can make sense to guard against that - but the `check_guess_hit` is so small it's verifiable that it can't return anything other than those 2 states, or raise an Exception, which wouldn't be caught by the if block anyway.